core/src/main/java/org/apache/calcite/sql/SqlKind.java

```diff
@@ -361,6 +361,9 @@
   /** {@code LIKE} operator. */
   LIKE,
 
+  /** {@code NOT LIKE} operator. */
+  NOT_LIKE("NOT LIKE"),
```

Comment:
Why is this necessary? Why can't not be handled as ~(like). If that because the some/any logic needs to be be fixed.

not all -> any like
not any -> not(any like)